### PR TITLE
Update instructions in README.docker.md

### PR DESCRIPTION
**Pull request by gshaw-pivotal**
- Include how to make Greenplum within docker accessible to SQL editors
  running on the local machine (outside of docker).
- Update the pip install command so that psutil and lockfile are
  accessible when the make cluster command is executed.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@gshaw-pivotal Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=6604)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=6604) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@gshaw-pivotal Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=6604)!

File: README.docker.md [hunk 1]
```diff
@@ -120,6 +125,14 @@ source gpdemo-env.sh
 psql postgres
 ```
 
+Modify the `pg_hba.conf` file located in `/home/gpadmin/gpdb_src/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/` to include the following:
```
Comment (jimmyyih) [inline README.docker.md]: Assuming gpdemo cluster's `gpdemo-env.sh` has been sourced prior (or user exported `PGPORT` and `MASTER_DATA_DIRECTORY` from `gpinitsystem` output), we should change the hardcoded path here to $MASTER_DATA_DIRECTORY.

**Comment by Eulerizeit** [comment]
@jimmyyih Gavin doesn't have commit rights. Can you fix and commit this please?

File: README.docker.md [hunk 2]
```diff
@@ -120,6 +125,14 @@ source gpdemo-env.sh
...
+
+`host all gpadmin 0.0.0.0 trust`
+
+This instructs Greenplum within docker to trust all connections from the local machine, like that from an SQL editor.
+
+Then run `gpstop -r` to restart the Greenplum cluster so that the changes in `pg_hba.conf` take effect.
```
[the remark this answers is not in the document]
Comment (gshaw-pivotal) [reply README.docker.md]: Modified to now include `gstop -au`.

File: README.docker.md (hunk 1 above)
Comment (gshaw-pivotal) [reply README.docker.md]: Modified to the `MASTER_DATA_DIRECTORY` and added a note that this assumes that `gpdemo-env.sh` has been sourced.

**Comment by gshaw-pivotal** [comment]
Moved the `pip install --user psutil lockfile` command to after gpadmin user creation.

**Pull request merged**